### PR TITLE
Intercept WinForms properties

### PR DESCRIPTION
**Pull request by melytc**
WinForms project have specific Applcation Framework properties that need to be read and saved in the corresponding Application.myapp file. For now, we will intercept those properties via the `ApplicationFrameworkPropertiesValueProvider`, but ideally we would like to implement a `IProjectPropertiesProvider` to access the myapp file.

Addresses #7236.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/8252)

### CODE REVIEW

**Review by MiYanni — COMMENTED** [review]
I talked with Melissa about using an XmlSerializer instead and simply having an object model that represents the MyApp file. It'll do the job simpler and easier. It should also make it easier if we want to switch to the workflow that @tmeschter mentioned previously of using an in-memory representation of a document editor to read/write the file (instead of direct serialization to disk). It isn't necessary for the first implementation, though.

**Comment by melytc** [comment]
My initial thought about the two approaches is that it depends on what we value more: I see how saving/modifying properties as needed (per property) can save us memory but be more time consuming; in the other hand, having a model with the expected structure (and maybe save a dictionary with everything else that could exist there) with an approach of saving everything in this model will cost more memory but be time efficient.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkPropertiesValueProvider.cs
Comment (melytc) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkPropertiesValueProvider.cs:101]: We discussed this in our 1:1 and addressed it in the last commit.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkPropertiesValueProvider.cs
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkPropertiesValueProvider.cs:128]: `InvalidOperationException` might be more appropriate than `InvalidDataException`. The latter is intended for invalid data in streams. Not a big deal though.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs:67]: You should decide one way or another about this before merging.

**Review by drewnoakes — APPROVED** [review]
Nice!

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkPropertiesValueProvider.cs
Comment (melytc) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkPropertiesValueProvider.cs]: Good point, addressed in the next commit.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs
Comment (melytc) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs]: Decided to do a try/catch when creating the DocData, and return null when we can't create it.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkPropertiesValueProvider.cs
Comment (MiYanni) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkPropertiesValueProvider.cs:107]: nit: Commented code.

**Review by MiYanni — APPROVED** [review]

**Pull request merged**
